irods_specific_implementation/irods_rest_stream_get_api_implementation.h

```diff
@@ -1,82 +1,85 @@
-
 #include "irods_rest_api_base.h"
 
 #include "dstream.hpp"
+#include "pistache/optional.h"
 #include "transport/default_transport.hpp"
+#include "rodsErrorTable.h"
+#include "irods_exception.hpp"
+#include "ticketAdmin.h"
+
+#include "pistache/http.h"
 
 // this is contractually tied directly to the swagger api definition, and the below implementation
 #define MACRO_IRODS_STREAM_GET_API_IMPLEMENTATION \
     Pistache::Http::Code code; \
     std::string message; \
-    std::tie(code, message) = irods_stream_get_(headers.getRaw("Authorization").value(), body, path.get(), offset.get(), limit.get()); \
+    std::tie(code, message) = irods_stream_get_(headers, body, path, offset, limit); \
     response.send(code, message);
 
-namespace irods::rest {
-
+namespace irods::rest
+{
     // this is contractually tied directly to the api implementation
     const std::string service_name{"irods_rest_cpp_stream_get_server"};
 
     namespace ix = irods::experimental;
     namespace fs = ix::filesystem;
     namespace io = ix::io;
 
-    class stream : public api_base {
-        public:
-            stream() : api_base{service_name}
-            {
-                // ctor
+    class stream : public api_base
+    {
+    public:
+        stream()
+            : api_base{service_name}
+        {
+            logger_->trace("Endpoint [{}] initialized.", service_name);
+        }
+
+        std::tuple<Pistache::Http::Code &&, std::string>
+        operator()(const Pistache::Http::Header::Collection& _headers,
+                   const std::string& _body,
+                   const std::string& _path,
+                   const Pistache::Optional<std::string>& _offset,
+                   const Pistache::Optional<std::string>& _limit) // TODO This shouldn't be optional
```

Comment:
Terrell agrees that this field should not be optional.

Reply:
@trel right now, we parse the limit/bytes_to_read as a 64-bit integer. should we place a cap on the maximum size of the buffer to fill?

Reply:
hmm, an admin configuration option?    with a default that's 'good enough' for known sane use cases...

Reply:
i'm wondering if admins need an option for this at all? do we know if there are clients who allocate 1Gb+ of memory for a single buffer read/write?

wouldn't the size of the buffer depend on the nature of the app and the size of the pipe between cpp rest and the iRODS server?

i'm starting to lean towards using a 32-bit int. with 32 bits, users have a wide range of buffer sizes available and at the same time, they don't run the risk of running out of memory.

here are the posts that have me thinking about this:
- https://stackoverflow.com/a/2811047/5818611
- https://stackoverflow.com/a/41747545/5818611

Reply:
I agree with that assessment.   This is a protection against blowing up the server memory.

Please leave comments to this effect in the defining code.

Reply:
done